src/resmom/mom_main.c

```diff
@@ -7092,8 +7092,12 @@ mom_over_limit(job *pjob)
 				retval = local_gettime(pres, &value);
 				if (retval != PBSE_NONE)
 					continue;
-				num = (u_long)((double)(time_now -
-					pjob->ji_qs.ji_stime) * wallfactor);
+				/* use the resources_used.walltime value */
+				retval = local_gettime(used, &num);
+				if (retval != PBSE_NONE)
+					continue;
+				/* add currently accumulating time */
+				num += time_now - pjob->ji_walltime_stamp;
```

Comment:
The original code showed:
num = (u_long)((double)(time_now - pjob->ji_qs.ji_stime) * wallfactor);
Does this really line up with the new code:
num += time_now - pjob->ji_walltime_stamp;
I'm worried about the accounting for the 'wallfactor'.

Reply:
The old method of counting wall time is `time_now - ji_stime`. This new method I am trying to introduce is accumulate wall time into `resources_used.walltime`. The `resources_used.walltime` value is obtained by `local_gettime(used, &num);`. Since it is only updated periodically, we need to add `time_now` minus `ji_walltime_stap` (which is the last time wall time was updated) to get the precise wall time.

Reply:
You are right about the `wallfactor` variable. I fixed it.